### PR TITLE
TypeORM wrapper to automatically detect 'trivial' identical records and not duplicate them

### LINKED ISSUE #195: Replace TypeORM query functionality with our own recursive-friendly one

### PR DESCRIPTION
**Pull request by dfellis**
Resolves #195

### CODE REVIEW

File: src/services/typeorm.ts [hunk 1]
```diff
@@ -50,8 +56,59 @@ export class TypeormWrapper {
...
+        }
+      }
+    }
+    // Now find the primary column of this entity (if one exists)
+    const primaryColumn = ormMetadata
+      .columns
+      .filter(c => c.target === entity)
+      .filter(c => c.options.primary)
+      .map(c => c.propertyName)
+      .shift();
```
Comment (dfellis) [inline src/services/typeorm.ts:80]: This manipulation of TypeORM's metadata may *seem* untyped, but TypeORM has fully typed these objects so the compiler will complain if you do the wrong thing with them, so changes to this metadata in TypeORM in the future should cause this code to fail to compile, and we can revisit how to adjust it without accidentally deploying broken code to production.

File: src/services/typeorm.ts [hunk 2]
```diff
@@ -50,8 +56,59 @@ export class TypeormWrapper {
...
     const repository = this.connection.manager.getRepository(entity);
+    if (value) {
+      if (Array.isArray(value)) {
+        for (const val of value) {
+          await this.maybeAttachPrimaryColumn(entity, val);
+        }
+      } else {
+        await this.maybeAttachPrimaryColumn(entity, value);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (dfellis) [reply src/services/typeorm.ts:111]: That's actually different. I twill run the `else` if `value` is not defined, I was intentionally avoiding that.

[the remark this answers is not in the document]
Comment (aguillenv) [reply src/services/typeorm.ts:111]: oh I ignored completely the `else`, my bad

**Comment by dfellis** [comment]
Abandoning this for now. If/when we run into it for real, we can recreate.

**Pull request closed without merging**